src/main/java/edu/kit/ipd/crowdcontrol/objectservice/moneytransfer/MoneyTransferManager.java

```diff
@@ -0,0 +1,172 @@
+package edu.kit.ipd.crowdcontrol.objectservice.moneytransfer;
+
+import edu.kit.ipd.crowdcontrol.objectservice.database.model.tables.records.GiftCodeRecord;
+import edu.kit.ipd.crowdcontrol.objectservice.database.model.tables.records.WorkerRecord;
+import edu.kit.ipd.crowdcontrol.objectservice.database.operations.WorkerBalanceOperations;
+import edu.kit.ipd.crowdcontrol.objectservice.database.operations.WorkerOperations;
+import edu.kit.ipd.crowdcontrol.objectservice.mail.MailHandler;
+import edu.kit.ipd.crowdcontrol.objectservice.template.Template;
+import org.jooq.Result;
+
+import java.io.*;
+import java.util.*;
+
+import javax.mail.Message;
+import javax.mail.MessagingException;
+
+/**
+ * Describes a Manager for money transfers. The Manager can log payments and pay off.
+ *
+ * @author Felix Rittler
+ */
+public class MoneyTransferManager {
+
+    MailHandler mailHandler;
+    WorkerBalanceOperations workerBalanceOperations;
+    WorkerOperations workerOperations;
+
+    int payOffThreshold;
+    int minGiftCodesCount;
+    String notificationMailAddress;
+    StringBuilder notificationText;
+
+
+    public MoneyTransferManager(MailHandler mailHandler, WorkerBalanceOperations workerBalanceOperations, WorkerOperations workerOperations, String notificationMailAddress) throws MessagingException {
+        this.mailHandler = mailHandler;
+        this.workerOperations = workerOperations;
+        this.workerBalanceOperations = workerBalanceOperations;
+        this.minGiftCodesCount = 10;
```

Comment:
eeeeer I dont like that at all! maybe some dynamic cacluation ? lets say you have 10 times 1€, this is enough for a few weeks (under the assumation that only 10 workers with 1€ payload appear). What about calulation the number of outgoing codes from the last day + a few , and a minimum of 10 ? Or capture if a experiment is in creative stop, and if it is prepare to have enough money (optimistic caclulation since you dont know the bonuses)